transmission_interface/test/differential_transmission_loader_test.cpp

```diff
@@ -0,0 +1,469 @@
+
+  // Validate transmission
+  transmission_interface::DifferentialTransmission * differential_transmission =
+    dynamic_cast<transmission_interface::DifferentialTransmission *>(transmission.get());
+  ASSERT_TRUE(nullptr != differential_transmission);
+
+  const std::vector<double> & actuator_reduction =
+    differential_transmission->get_actuator_reduction();
+  EXPECT_EQ(50.0, actuator_reduction[0]);
+  EXPECT_EQ(-50.0, actuator_reduction[1]);
+
+  const std::vector<double> & joint_reduction = differential_transmission->get_joint_reduction();
+  EXPECT_EQ(2.0, joint_reduction[0]);
+  EXPECT_EQ(-2.0, joint_reduction[1]);
+
+  const std::vector<double> & joint_offset = differential_transmission->get_joint_offset();
+  EXPECT_EQ(0.5, joint_offset[0]);
+  EXPECT_EQ(-0.5, joint_offset[1]);
+}
+
+TEST(DifferentialTransmissionLoaderTest, only_mech_red_specified)
+{
+  // Parse transmission info
+  std::string urdf_to_test = R"(
+    <?xml version="1.0"?>
+    <robot name="robot" xmlns="http://www.ros.org">
+      <ros2_control name="FullSpec" type="system">
+        <joint name="joint1">
+          <command_interface name="velocity">
+            <param name="min">-0.5</param>
+            <param name="max">0.5</param>
+          </command_interface>
+          <state_interface name="velocity"/>
+        </joint>
+        <joint name="joint2">
+          <command_interface name="position">
+            <param name="min">-1</param>
+            <param name="max">1</param>
+          </command_interface>
+          <state_interface name="position"/>
+        </joint>
+        <transmission name="transmission1">
+          <plugin>transmission_interface/DifferentialTransmission</plugin>
+          <actuator name="joint1_motor" role="actuator1">
+            <mechanical_reduction>50</mechanical_reduction>
+          </actuator>
+          <actuator name="joint2_motor" role="actuator2">
+            <mechanical_reduction>-50</mechanical_reduction>
+          </actuator>
+          <joint name="joint1" role="joint1">
+            <mechanical_reduction>1.0</mechanical_reduction>
+          </joint>
+          <joint name="joint2" role="joint2">
+            <mechanical_reduction>1.0</mechanical_reduction>
+          </joint>
+        </transmission>
+      </ros2_control>
+    </robot>
+    )";
+  std::vector<hardware_interface::HardwareInfo> infos =
+    hardware_interface::parse_control_resources_from_urdf(urdf_to_test);
+  ASSERT_THAT(infos[0].transmissions, SizeIs(1));
+
+  // Transmission loader
+  TransmissionPluginLoader loader;
+  std::shared_ptr<transmission_interface::TransmissionLoader> transmission_loader =
+    loader.create(infos[0].transmissions[0].type);
+  ASSERT_TRUE(nullptr != transmission_loader);
+
+  std::shared_ptr<transmission_interface::Transmission> transmission = nullptr;
+  const hardware_interface::TransmissionInfo & info = infos[0].transmissions[0];
+  transmission = transmission_loader->load(info);
```

Comment:
Aren't we throwing an exception here in `SimpleTransmission`? If so we should do the same.